repo/fsrepo/fsrepo.go

```diff
@@ -482,6 +482,14 @@ func (r *FSRepo) SetConfigKey(key string, value interface{}) error {
 		return err
 	}
 
+	// Load private key to guard against it being overwritten.
+	// NOTE: this is a temporary measure to secure this field until we move
+	// keys out of the config file.
+	pkval, err := common.MapGetKV(mapconf, "Identity.PrivKey")
```

Comment:
I would merge a PR that cleanly moves the private key out of the config if
paired with a migration. I do not have time to work on that.

On Mon, Aug 29, 2016, 16:41 Juan Benet <notifications@github.com> wrote:

> In repo/fsrepo/fsrepo.go
> <https://github.com/ipfs/go-ipfs/pull/3147#discussion_r76708163>:
>
> > @@ -482,6 +482,14 @@ func (r *FSRepo) SetConfigKey(key string, value interface{}) error {
> >  		return err
> >  	}
> >
> > +	// Load private key to guard against it being overwritten.
> > +	// NOTE: this is a temporary measure to secure this field until we move
> > +	// keys out of the config file.
> > +	pkval, err := common.MapGetKV(mapconf, "Identity.PrivKey")
>
>
>    - What if the name of the fields in the structs changes and people
>    forget to change it here? should be a const.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ipfs/go-ipfs/pull/3147/files/d290d567fc5ced87bb6dd7634d80b697272d42ff#r76708163>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABL4HGKfwgzdeG15CckDMgcf7McbFFyFks5qk24TgaJpZM4Jv9id>
> .
>